Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -17,79 +23,127 @@
+
+    @Override
+    public ITransformedNetwork getTransformedNetwork(String networkId, String versionID)
+            throws QuadrigaStorageException {
+
+        ITransformedNetwork networkJSon = null;
+        List<INetworkNodeInfo> oldNetworkTopNodesList = null;
+
+        try {
+            oldNetworkTopNodesList = networkManager.getNetworkTopNodesByVersion(networkId, Integer.parseInt(versionID));
+        } catch (QuadrigaStorageException e) {
+            logger.error("DB Error while getting network top nodes", e);
+        }
+
+        networkJSon = transformer.transformNetwork(oldNetworkTopNodesList);
+        return networkJSon;
+    }
+
+    @Override
+    public ITransformedNetwork getTransformedNetwork(String networkId) throws QuadrigaStorageException {
+
+        ITransformedNetwork networkJSon = null;
+
+        List<INetworkNodeInfo> networkTopNodesList = null;
+
+        try {
+            networkTopNodesList = networkManager.getNetworkTopNodes(networkId);
+        } catch (QuadrigaStorageException e) {
+            logger.error("DB Error while getting network top nodes", e);
+            return null;
+        }
+
+        networkJSon = transformer.transformNetwork(networkTopNodesList);
+
+        return networkJSon;
+    }
+
+    @Override
+    public ITransformedNetwork getTransformedNetworkOfProject(String projectId)
+            throws QuadrigaStorageException {
+
+        List<INetwork> networkList = getNetworkList(projectId);
+
+        if (networkList == null) {
+            return null;
+        }
+
+        List<INetworkNodeInfo> networkNodeInfoList = new ArrayList<INetworkNodeInfo>();
+        for (INetwork network : networkList) {
+            try {
+                List<INetworkNodeInfo> localNetworkNodeInfoList =
+                        networkManager.getNetworkTopNodes(network.getNetworkId());
+                if (localNetworkNodeInfoList != null) {
+                    networkNodeInfoList.addAll(localNetworkNodeInfoList);
+                }
+            } catch (QuadrigaStorageException e) {
+                throw new QuadrigaStorageException("Database Error while getting network top nodes for a network with ID: " +
+                        network.getNetworkId(), e);
+            }
+        }
+
+        ITransformedNetwork transformedNetwork = transformer.transformNetwork(networkNodeInfoList);
+
+        // combine all the nodes except predicate nodes
+
+        Map<String, Node> nodes = transformedNetwork.getNodes();
+        List<Link> links = transformedNetwork.getLinks();
+
+        Map<String, Node> updatedNodes = new HashMap<String, Node>();
+        int index = 0;
+        for (Node node: nodes.values()) {
+            // if the node is predicate node
+            // add to the updated nodes map
+            if (node instanceof PredicateNode) {
+                // the key does not matter
+                // as it is never being used
+                updatedNodes.put(String.valueOf(++index), node);
```

Comment:
Is there a reason you create a new key here instead of using the one that already exists (the one in the nodes map)?